## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 448928`
Commit: [`1cc8dc267f1d1265cf53727f5f9147d3fe2dfef0`](https://github.com/NixOS/nixpkgs/commit/1cc8dc267f1d1265cf53727f5f9147d3fe2dfef0) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/1cc8dc267f1d1265cf53727f5f9147d3fe2dfef0..pull/448928/head))
Merge: [`8ac663a3f3ef181c71c0cb505a222e22a6a48e22`](https://github.com/NixOS/nixpkgs/commit/8ac663a3f3ef181c71c0cb505a222e22a6a48e22)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/18266440568


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>freedink</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>freedink</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>freedink</summary>
<pre>In file included from IOGfxDisplay.cpp:2:
../config.h:941:5: warning: option &#x27;8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wmissing-prototypes-Wmissing-prototypes8;;&#x27; is valid for C/ObjC but not for C++ [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-pragmas-Wpragmas8;;]
  941 |     _Pragma (&quot;GCC diagnostic ignored \&quot;-Wmissing-prototypes\&quot;&quot;) \
      |     ^~~~~~~
make[2]: *** [Makefile:2101: ImageLoader.o] Error 1
make[2]: *** Waiting for unfinished jobs....
hardness_tiles.cpp: In function &#x27;void load_hard()&#x27;:
hardness_tiles.cpp:105:16: warning: ignoring return value of &#x27;size_t fread(void*, size_t, size_t, FILE*)&#x27; declared with attribute &#x27;warn_unused_result&#x27; [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result8;;]
  105 |           fread(&amp;hmap.htile[i].hm[x][y], 1, 1, f);
      |           ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hardness_tiles.cpp:107:12: warning: ignoring return value of &#x27;size_t fread(void*, size_t, size_t, FILE*)&#x27; declared with attribute &#x27;warn_unused_result&#x27; [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result8;;]
  107 |       fread(skipbuf, 2, 1, f); // reproduce memory alignment
      |       ~~~~~^~~~~~~~~~~~~~~~~~
hardness_tiles.cpp:108:12: warning: ignoring return value of &#x27;size_t fread(void*, size_t, size_t, FILE*)&#x27; declared with attribute &#x27;warn_unused_result&#x27; [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result8;;]
  108 |       fread(skipbuf, 4, 1, f); // unused &#x27;hold&#x27; field
      |       ~~~~~^~~~~~~~~~~~~~~~~~
make[2]: Leaving directory &#x27;/build/freedink-109.6/src&#x27;
make[1]: *** [Makefile:1026: all-recursive] Error 1
make[1]: Leaving directory &#x27;/build/freedink-109.6&#x27;
make: *** [Makefile:967: all] Error 2</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>freedink</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>freedink</li>
  </ul>
</details>

--- a/pkgs/by-name/fr/freedink/package.nix
+++ b/pkgs/by-name/fr/freedink/package.nix
@@ -1,65 +1,106 @@
 {
-  lib,
-  stdenv,
+  cxxtest,
   fetchurl,
-  SDL,
-  SDL_mixer,
-  SDL_image,
-  SDL_ttf,
-  SDL_gfx,
-  pkg-config,
-  intltool,
   fontconfig,
+  glm,
+  help2man,
+  intltool,
+  lib,
   libzip,
+  pkg-config,
+  SDL2,
+  SDL2_gfx,
+  SDL2_image,
+  SDL2_mixer,
+  SDL2_ttf,
+  stdenv,
   zip,
   zlib,
 }:
 
 let
-  version = "1.08.20121209";
 
-  freedink_data = stdenv.mkDerivation rec {
+  freedink_data = stdenv.mkDerivation (finalAttrs: {
     pname = "freedink-data";
-    inherit version;
+    version = "1.08.20190120";
 
     src = fetchurl {
-      url = "mirror://gnu/freedink/${pname}-${version}.tar.gz";
-      sha256 = "1mhns09l1s898x18ahbcy9gabrmgsr8dv7pm0a2ivid8mhxahn1j";
+      url = "mirror://gnu/freedink/freedink-data-${finalAttrs.version}.tar.gz";
+      hash = "sha256-cV9EdzsFtzqeybYrDhUvPygb4aFRL7qqOGF22pTP+50=";
     };
 
-    prePatch = "substituteInPlace Makefile --replace /usr/local $out";
-  };
+    prePatch = "substituteInPlace Makefile --replace-fail /usr/local $out";
+  });
 
 in
-stdenv.mkDerivation rec {
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "freedink";
-  inherit version;
+  version = "109.6";
 
   src = fetchurl {
-    url = "mirror://gnu/freedink/${pname}-${version}.tar.gz";
-    sha256 = "19xximbcm6506kvpf3s0q96697kmzca3yrjdr6dgphklp33zqsqr";
+    url = "mirror://gnu/freedink/freedink-${finalAttrs.version}.tar.gz";
+    hash = "sha256-Xgs1rI9G17uH5lbv1fnHwqwabFGakI/FtYHlJleYEAI=";
   };
 
   nativeBuildInputs = [
-    pkg-config
+    cxxtest
+    help2man
     intltool
+    pkg-config
   ];
 
   buildInputs = [
-    SDL
-    SDL_mixer
-    SDL_image
-    SDL_ttf
-    SDL_gfx
     fontconfig
+    glm
     libzip
+    SDL2
+    SDL2_mixer
+    SDL2_image
+    SDL2_ttf
+    SDL2_gfx
     zip
     zlib
   ];
 
-  preConfigure = ''
-    # Build fails on Linux with windres.
-    export ac_cv_prog_ac_ct_WINDRES=
+  preBuild = ''
+        # Fix SDL2_ttf constness error in gfx_fonts.cpp
+        substituteInPlace src/gfx_fonts.cpp \
+          --replace-fail "char *familyname = TTF_FontFaceFamilyName(font);" \
+                    "const char *familyname = TTF_FontFaceFamilyName(font);"
+        substituteInPlace src/gfx_fonts.cpp \
+          --replace-fail "char *stylename = TTF_FontFaceStyleName(font);" \
+                    "const char *stylename = TTF_FontFaceStyleName(font);"
+
+        # Fix missing SDL hint macro error in input.cpp
+        substituteInPlace src/input.cpp \
+          --replace-fail "SDL_SetHint(SDL_HINT_ANDROID_SEPARATE_MOUSE_AND_TOUCH, \"0\");" \
+                    "/* SDL_SetHint(SDL_HINT_ANDROID_SEPARATE_MOUSE_AND_TOUCH, \"0\"); */"
+
+        # Fix config.h errors in some platforms
+        substituteInPlace src/ImageLoader.cpp \
+          --replace-fail '#include "ImageLoader.h"' '#include "../config.h"
+    #include "ImageLoader.h"'
+
+        substituteInPlace src/IOGfxGLFuncs.cpp \
+          --replace-fail '#include "IOGfxGLFuncs.h"' '#include "../config.h"
+    #include "IOGfxGLFuncs.h"'
+
+        substituteInPlace src/IOGfxSurface.cpp \
+          --replace-fail '#include "IOGfxSurface.h"' '#include "../config.h"
+    #include "IOGfxSurface.h"'
+
+        substituteInPlace src/IOGfxSurfaceSW.cpp \
+          --replace-fail '#include "IOGfxSurfaceSW.h"' '#include "../config.h"
+    #include "IOGfxSurfaceSW.h"'
+
+        substituteInPlace src/IOGfxPrimitivesSW.cpp \
+          --replace-fail '#include "SDL.h"' '#include "../config.h"
+    #include "SDL.h"'
+
+        substituteInPlace src/dinkc_console_renderer.cpp \
+          --replace-fail '#include "dinkc_console_renderer.h"' '#include "../config.h"
+    #include "dinkc_console_renderer.h"'
   '';
 
   postInstall = ''
@@ -71,18 +112,15 @@ stdenv.mkDerivation rec {
 
   meta = {
     description = "Free, portable and enhanced version of the Dink Smallwood game engine";
-
     longDescription = ''
       GNU FreeDink is a new and portable version of the Dink Smallwood
       game engine, which runs the original game as well as its D-Mods,
       with close compatibility, under multiple platforms.
     '';
-
-    homepage = "https://www.gnu.org/software/freedink/"; # Formerly http://www.freedink.org
+    homepage = "https://gnu.org/software/freedink/"; # Formerly http://www.freedink.org
     license = lib.licenses.gpl3Plus;
-
     maintainers = [ ];
     platforms = lib.platforms.all;
-    hydraPlatforms = lib.platforms.linux; # sdl-config times out on darwin
+    mainProgram = "freedink";
   };
-}
+})

--- a/pkgs/by-name/fr/freedink/package.nix
+++ b/pkgs/by-name/fr/freedink/package.nix
@@ -119,7 +119,7 @@ stdenv.mkDerivation (finalAttrs: {
     '';
     homepage = "https://gnu.org/software/freedink/"; # Formerly http://www.freedink.org
     license = lib.licenses.gpl3Plus;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ iedame ];
     platforms = lib.platforms.all;
     mainProgram = "freedink";
   };

